quantecon/game_theory/repeated_game.py

```diff
@@ -0,0 +1,454 @@
+"""
+Tools for repeated game.
+
+"""
+
+import numpy as np
+from scipy.spatial import ConvexHull
+from numba import jit, njit
+
+
+class RepeatedGame:
+    """
+    Class representing an N-player repeated game.
+
+    Parameters
+    ----------
+    stage_game : NormalFormGame
+                 The stage game used to create the repeated game.
+
+    delta : scalar(float)
+            The common discount rate at which all players discount the future.
+
+    """
+    def __init__(self, stage_game, delta):
+        self.sg = stage_game
+        self.delta = delta
+        self.N = stage_game.N
+        self.nums_actions = stage_game.nums_actions
+
+    def equilibrium_payoffs(self, method=None, options=None):
+        """
+        Compute the set of payoff pairs of all pure-strategy subgame-perfect
+        equilibria with public randomization for any repeated two-player games
+        with perfect monitoring and discounting.
+
+        Parameters
+        ----------
+        method : str, optional
+            The method for solving the equilibrium payoff set.
+
+        options : dict, optional
+            A dictionary of method options. For example, 'abreu_sannikov'
+            method accepts the following options:
+
+                tol : scalar(float)
+                    Tolerance for convergence checking.
+                max_iter : scalar(int)
+                    Maximum number of iterations.
+                u_init : ndarray(float, ndim=1)
+                    The initial guess of threat points.
+
+        Notes
+        -----
+        Here lists all the implemented methods. The default method
+        is 'abreu_sannikov'.
+
+            1. 'abreu_sannikov'
+        """
+        if method is None:
+            method = 'abreu_sannikov'
+
+        if options is None:
+            options = {}
+
+        if method in ('abreu_sannikov', 'AS'):
+            return _equilibrium_payoffs_abreu_sannikov(self, **options)
+        else:
+            msg = f"method {method} not supported."
+            raise NotImplementedError(msg)
```

Comment:
I would expect a `ValueError` if I pass `method='something_nonsense'`.

Reply:
That's True. I shall change this exception.

Considering that 'outerapproximation' method is going to be added in the future, shall we set a separate `NotImplementedError` exception for it?